core/src/main/scala/kafka/server/ReplicaManager.scala

```diff
@@ -1384,7 +1384,13 @@ class ReplicaManager(val config: KafkaConfig,
     for ((dir, reps) <- replicasByDir) {
       val hwms = reps.map(r => r.topicPartition -> r.highWatermark.messageOffset).toMap
       try {
-        highWatermarkCheckpoints.get(dir).foreach(_.write(hwms))
+        highWatermarkCheckpoints.get(dir) match {
```

Comment:
Nit:

```scala
highWatermarkCheckpoints.get(dir).foreach { checkpointFile =>
  ...
}
```

Reply:
Yeah that is better. I have updated the patch as suggested. Thanks for the suggestion.